spec/researches/keywordCountSpec.js

```diff
@@ -2,41 +2,113 @@
-		mockPaper = new Paper( "a string of text with the key-word in it, density should be 0.0%", { keyword: "key word" } );
-		expect( keywordCount( mockPaper ).count ).toBe( 0 );
-		mockPaper = new Paper( "a string of text with the key&word in it, density should be 7.7%", { keyword: "key&word" } );
-		expect( keywordCount( mockPaper ).count ).toBe( 1 );
-		mockPaper = new Paper( "<img src='http://image.com/image.png'>", { keyword: "key&word" } );
-		expect( keywordCount( mockPaper ).count ).toBe( 0 );
-		// Consecutive keywords are skipped, so this will match 2 times.
-		mockPaper = new Paper( "This is a nice string with a keyword keyword keyword.", { keyword: "keyword" } );
-		expect( keywordCount( mockPaper ).count ).toBe( 2 );
-		mockPaper = new Paper( "a string of text with the $keyword in it, density should be 7.7%", { keyword: "$keyword" } );
-		expect( keywordCount( mockPaper ).count ).toBe( 1 );
-		mockPaper = new Paper( "a string of text with the Keyword in it, density should be 7.7%", { keyword: "keyword" } );
-		expect( keywordCount( mockPaper ).count ).toBe( 1 );
-		mockPaper = new Paper( "a string of text with the Key word in it, density should be 7.14%", { keyword: "key word" } );
-		expect( keywordCount( mockPaper ).count ).toBe( 1 );
-		mockPaper = new Paper( "a string with quotes to match the key'word, even if the quotes differ", { keyword: "key’word" } );
-		expect( keywordCount( mockPaper ).count ).toBe( 1 );
+	it( "counts a string of text with a keyword in it.", function() {
+		const mockPaper = new Paper( "a string of text with the keyword in it, density should be 7.7%" );
+		expect( keywordCount( mockPaper, mockResearcher ).count ).toBe( 1 );
+	} );
+
+	it( "counts a string of text with no keyword in it.", function() {
+		const mockPaper = new Paper( "a string of text, density should be 0.0%" );
+		expect( keywordCount( mockPaper, mockResearcher ).count ).toBe( 0 );
+	} );
+
+	it( "counts a string of text with multiple occurrences of the keyword in it.", function() {
+		const mockPaper = new Paper( "a string of text with the key word in it, with more key words." );
+		expect( keywordCount( mockPaper, mockResearcherKeyWord ).count ).toBe( 2 );
+	} );
+
+	it( "counts a string of text with german diacritics and eszett as the keyword", function() {
+		const mockPaper = new Paper( "Waltz keepin auf mitz auf keepin äöüß weiner blitz deutsch spitzen." );
+		expect( keywordCount( mockPaper, mockResearcherGermanDiacritics ).count ).toBe( 1 );
+	} );
+
+	it( "counts a string with multiple keyword morphological forms", function() {
+		const mockPaper = new Paper( "A string of text with a keyword and multiple keywords in it." );
+		expect( keywordCount( mockPaper, mockResearcher ).count ).toBe( 2 );
+	} );
+
+	it( "counts a string with multiple keywords", function() {
+		const mockPaper = new Paper( "A string with strawberries and a keyword." );
+		expect( keywordCount( mockPaper, mockResearcherMultipleKeywords ).count ).toBe( 1 );
```

Comment:
Why is keyword count 1 and not 2? Shouldn't both `strawberries` and `keyword` be matched?

Reply:
Actually `strawberries` and `keyword` here would be two words of the same keyphrase, so the count is okay. (I've removed this test since it's a duplicate of another test condition.)